xgi/drawing/layout.py

```diff
@@ -99,7 +100,7 @@ def barycenter_spring_layout(H, return_phantom_graph=False):
         G.add_edge(i, j)
     
     # Adding phantom nodes and connections therein
-    phantom_node_id = H.number_of_nodes() #the first id is N (the first node is 0)
+    phantom_node_id = max(H.nodes)+1 
```

Comment:
H.num_nodes

Reply:
What if nodes are not numbered from 0 to N-1? Let's say your nodes are {1,2,4}, so the number of nodes is 3, then H.num_nodes+1 (the ID of the phantom node) will be 4 which is already in use.
I'm open to better solutions :)

Reply:
Ah yea good point, my bad :) Is it possible that an edge already ID `H.num_nodes+1' and that it clashes? I don't remember how we deal with IDs from nodes and edges. In any case, forget my suggestion hehe.